python/lsst/pipe/base/pipeline_graph/visualization/_formatting.py

```diff
@@ -0,0 +1,175 @@
+    options : `NodeAttributeOptions`
+        Options for how much information to display.
+    width : `int` or `None`
+        Number of display columns that node text can occupy.  `None` for
+        unlimited.
+    """
+
+    def __init__(
+        self,
+        xgraph: networkx.DiGraph | networkx.MultiDiGraph,
+        options: NodeAttributeOptions,
+        width: int | None,
+    ):
+        self.xgraph = xgraph
+        self.options = options
+        self.width = width
+        self.deferred: list[tuple[str, tuple[str, str], list[str]]] = []
+
+    def __call__(self, node: DisplayNodeKey, x: int, style: tuple[str, str]) -> str:
+        state = self.xgraph.nodes[node]
+        terms: list[str] = [f"{node}:" if self.options else str(node)]
+        if self.options.dimensions and node.node_type != NodeType.TASK_INIT:
+            terms.append(self.format_dimensions(state["dimensions"]))
+        if (
+            self.options.task_classes
+            and node.node_type is NodeType.TASK
+            or node.node_type is NodeType.TASK_INIT
+        ):
+            terms.append(self.format_task_class(state["task_class_name"]))
+        if self.options.storage_classes and node.node_type is NodeType.DATASET_TYPE:
+            terms.append(state["storage_class_name"])
+        description = " ".join(terms)
+        if self.width and len(description) > self.width:
+            index = f"[{len(self.deferred) + 1}]"
+            self.deferred.append((index, style, terms))
+            return f"{description[:self.width - len(index) - 6]}...{style[0]}{index}{style[1]} "
+        return description
+
+    def format_dimensions(self, dimensions: DimensionGraph) -> str:
+        """Format the dimensions of a task or dataset type node."""
+        match self.options.dimensions:
+            case "full":
+                return str(dimensions)
+            case "concise":
+                kept = set(dimensions)
+                done = False
+                while not done:
+                    for dimension in kept:
+                        if any(dimension != other and dimension in other.graph for other in kept):
+                            kept.remove(dimension)
+                            break
+                    else:
+                        done = True
+                # We still iterate over dimensions instead of kept to preserve
+                # order.
+                return f"{{{', '.join(d.name for d in dimensions if d in kept)}}}"
+            case False:
+                return ""
+        raise ValueError(f"Invalid display option for dimensions: {self.options.dimensions!r}.")
+
+    def format_task_class(self, task_class_name: str) -> str:
+        """Format the type object for a task or task init node."""
+        match self.options.task_classes:
+            case "full":
+                return task_class_name
+            case "concise":
+                return task_class_name.split(".")[-1]
+            case False:
```

Comment:
I see that this is covered by the tests, so it must be correct. But I am curious why this is `False` instead of `None`?

Reply:
I reserved `None`  to mean something like "automatic", i.e. "turn this on or off based on other context-dependent information", so I went with `False` as the "definitely don't display this information" option.  But by the time the code gets here any `None` value will have been replaced by one of the other options.